crates/extricrate/src/lib.rs

```diff
@@ -355,15 +429,90 @@ pub mod dependencies {
                     statement_type: UseStatementType::Simple("HashMap".to_owned()),
                 }]
             );
+            assert_eq!(
+                module_b_statement.source_module,
+                "crate::module_a::module_b".into()
+            );
+            assert_eq!(module_b_statement.target_modules, vec!["foo".into()]);
+            assert_eq!(
+                module_b_statement.statement.span.start(),
+                LineColumn { line: 1, column: 0 }
+            );
+            assert_eq!(
+                module_b_statement.statement.span.end(),
+                LineColumn { line: 1, column: 8 }
+            );
+            assert_eq!(
+                module_b_statement.statement.items,
+                vec![NormalizedUseStatement {
+                    module_name: "".into(),
+                    statement_type: UseStatementType::Simple("foo".to_owned()),
+                }]
+            );
+        }
+
+        #[test]
+        fn gets_a_nested_dependency() {
+            let test_project = Path::new(env!("CARGO_MANIFEST_DIR")).join("tests/fixtures/inline/");
+            let res = list_use_statements(&test_project).expect("Failed to list statements");
+
+            let main_statement = &res.get(&File("src/main.rs".to_owned())).unwrap()[0];
+            let nested_statement = &res.get(&File("src/main.rs".to_owned())).unwrap()[1];
+            assert_eq!(main_statement.source_module, "crate::module_a".into());
+            assert_eq!(
+                main_statement.target_modules,
+                vec!["crate::module_a::module_b".into()]
+            );
+            assert_eq!(
+                main_statement.statement.span.start(),
+                LineColumn { line: 2, column: 4 }
+            );
+            assert_eq!(
+                main_statement.statement.span.end(),
+                LineColumn {
+                    line: 2,
+                    column: 34
+                }
+            );
+            assert_eq!(
+                main_statement.statement.items,
+                vec![NormalizedUseStatement {
+                    module_name: "crate::module_a".into(),
+                    statement_type: UseStatementType::Simple("module_b".to_owned()),
+                }]
+            );
+            assert_eq!(
+                nested_statement.source_module,
+                "crate::module_a::module_b".into()
+            );
+            assert_eq!(nested_statement.target_modules, vec!["foo::Bar".into()]);
```

Comment:
Shouldn't that be:
```suggestion
            assert_eq!(nested_statement.target_modules, vec!["foo".into()]);
```

We might have to use the casing to know whether an element is a module or an item.

Reply:
good catch, this was definitely due to my misunderstanding of what target_modules should have been 😄 should be behaving as you expect now!